pkg/cloudprovider/providers/azure/azure_instance_metadata.go

```diff
@@ -61,28 +54,43 @@ type Subnet struct {
 	Prefix  string `json:"prefix"`
 }
 
+type InstanceMetadata struct {
+	baseURL string
+}
+
+func NewInstanceMetadata() *InstanceMetadata {
+	return &InstanceMetadata{
+		baseURL: metadataURL,
+	}
+}
+
+// makeMetadataURL makes a complete metadata URL from the given path.
+func (i *InstanceMetadata) makeMetadataURL(path string) string {
+	return i.baseURL + path
+}
+
 // QueryMetadataJSON queries the metadata server and populates the passed in object
-func QueryMetadataJSON(path string, obj interface{}) error {
-	data, err := queryMetadataBytes(path, "json")
+func (i *InstanceMetadata) QueryMetadataJSON(path string, obj interface{}) error {
+	data, err := i.queryMetadataBytes(path, "json")
 	if err != nil {
 		return err
 	}
 	return json.Unmarshal(data, obj)
 }
 
 // QueryMetadataText queries the metadata server and returns the corresponding text
-func QueryMetadataText(path string) (string, error) {
-	data, err := queryMetadataBytes(path, "text")
+func (i *InstanceMetadata) QueryMetadataText(path string) (string, error) {
+	data, err := i.queryMetadataBytes(path, "text")
 	if err != nil {
 		return "", err
 	}
 	return string(data), err
 }
 
-func queryMetadataBytes(path, format string) ([]byte, error) {
+func (i *InstanceMetadata) queryMetadataBytes(path, format string) ([]byte, error) {
```

Comment:
One more comment nit, just to be consistent (cf: the `// makeMetadataURL` comment)